dispatches/models/fossil_case/properties/hitecsalt_properties.py

```diff
@@ -0,0 +1,424 @@
+            hold_state : flag indicating whether the initialization routine
+                         should unfix any state variables fixed during
+                         initialization (default=False).
+                         - True - states varaibles are not unfixed, and
+                                 a dict of returned containing flags for
+                                 which states were fixed during
+                                 initialization.
+                        - False - state variables are unfixed after
+                                 initialization by calling the
+                                 relase_state method
+        Returns:
+            If hold_states is True, returns a dict containing flags for
+            which states were fixed during initialization.
+        """
+        # Fix state variables if not already fixed by the control volume block
+        if state_vars_fixed is False:
+            # Fix state variables if not already fixed
+            Fflag = {}
+            Tflag = {}
+            Pflag = {}
+            
+            for k in blk.keys():
+                if blk[k].flow_mass.fixed is True:
+                    Fflag[k] = True
+                else:
+                    Fflag[k] = False
+                    if state_args is None:
+                        blk[k].flow_mass.fix()
+                    else:
+                        blk[k].flow_mass.fix(state_args["flow_mass"])
+
+                if blk[k].pressure.fixed is True:
+                    Pflag[k] = True
+                else:
+                    Pflag[k] = False
+                    if state_args is None:
+                        blk[k].pressure.fix()
+                    else:
+                        blk[k].pressure.fix(state_args["pressure"])
+
+                if blk[k].temperature.fixed is True:
+                    Tflag[k] = True
+                else:
+                    Tflag[k] = False
+                    if state_args is None:
+                        blk[k].temperature.fix()
+                    else:
+                        blk[k].temperature.fix(state_args["temperature"])
+
+            flags = {"Fflag": Fflag, "Tflag": Tflag, "Pflag": Pflag}
+
+        else:
+            # Check when the state vars are fixed already result in dof 0
+            for k in blk.keys():
+                if degrees_of_freedom(blk[k]) != 0:
+                    raise Exception("State vars fixed but degrees of freedom "
+                                    "for state block is not zero during "
+                                    "initialization.")
+
+        # Set solver options
+        if outlvl > 1:
+            stee = True
+        else:
+            stee = False
+
+        opt = get_solver(solver, optarg)
+
+        # ---------------------------------------------------------------------
+        # Solve property correlation
+        results = solve_indexed_blocks(opt, [blk], tee=stee)
+
+        if outlvl > 0:
```

Comment:
The logger should handle all of this, there is no need for `outlvl` here.

Reply:
addressed